apps/condo/domains/resident/schema/DiscoverServiceConsumersService.js

```diff
@@ -36,43 +47,98 @@ const DiscoverServiceConsumersService = new GQLCustomSchema('DiscoverServiceCons
 
-                const residents = await Resident.getAll(context,
-                    resident ? { id: resident.id, deletedAt: null } : {
-                        address: address,
-                        unitName: unitName,
-                        unitType: unitType,
-                        deletedAt: null,
-                    })
-
-                const billingAccounts = await BillingAccount.getAll(context,
-                    billingAccount ? { id: billingAccount.id, deletedAt: null } : {
-                        context: { status: 'Finished', deletedAt: null },
-                        property: { address: address, deletedAt: null  },
-                        unitName: unitName,
-                        unitType: unitType,
+                const reqId = get(context, ['req', 'id'])
+
+                const billingAccounts = await BillingAccount.getAll(
+                    context,
+                    {
                         deletedAt: null,
-                    })
+                        context: { status: CONTEXT_FINISHED_STATUS, deletedAt: null },
+                        ...billingAccount ? {
+                            id: billingAccount.id,
+                        } : {
+                            property: { address, deletedAt: null }, unitName, unitType,
+                        },
+                    },
+                )
 
+                const residents = await Resident.getAll(
+                    context,
+                    { deletedAt: null, ...resident ? { id: resident.id } : { address, unitName, unitType } },
+                )
 
-                const getPayload = (accountNumber, resident) => ({
-                    dv,
-                    sender,
-                    resident: { connect: { id: resident.id } },
-                    organization: { connect: { id: get(resident, ['organization', 'id']) } },
-                    accountNumber: accountNumber,
-                })
+                // Keep only residents of organizations the feature flag enabled for
+                const residentsFilteredByFeatureFlag = await asyncFilter(
+                    residents,
+                    async (resident) => await featureToggleManager.isFeatureEnabled(
+                        context,
+                        ENABLE_DISCOVER_SERVICE_CONSUMERS,
+                        { organization: resident.organization.id },
+                    ),
+                )
+
+                // Keep only residents of organizations the acquiring context is existing for
+                const residentsFilteredByOrganizationAcquiringContext = await asyncFilter(
+                    residentsFilteredByFeatureFlag,
+                    async (resident) => await AcquiringIntegrationContext.count(
+                        context,
+                        {
+                            deletedAt: null,
+                            organization: { id: get(resident, ['organization', 'id']) },
```

Comment:
1. `Undefined` is resolved to empty filter, use `null` as default value
2. resident.organization will not work for `SERVICE_PROVIDER`s, since residents are not linked to them

Reply:
Are `SERVICE_PROVIDER` even covered with tests?